ONSdigital/eq-survey-runner
# Add security headers

### Changes

This PR adds a number of security improvements to the survey runner app.
1. Adds secure http headers as recommended by OWASP
2. Pins all pypi dependencies to SHA256 hashes to stop downstream tampering
3. Sets all cookies to secure https use only.
### How to test
1. Download the OWASP zaproxy tool.
2. Run against a local version of the application.
3. Check that no cookie related warnings are given.
4. Run a clean install using `pip install -U -I  -r requirements.txt`
5. Edit one of the hashes in `requirements.txt`
6. Re-run the command `pip install -U -I  -r requirements.txt`
7. It should fail, warning of a tampering with one of the packages.
8. Set the environment variable DEV_MODE to false
9. Try and use the application locally - you should see the setting of cookies fail as they're not served over tls.
### Who can test

Anyone but @dhilton
